File: OrderProcessor/OrderProcessor.Tests/UnitTests/OrderValidatorTests.cs
using OrderProcessor.Models;
using OrderProcessor.Services;

namespace OrderProcessor.Tests.UnitTests;

public sealed class OrderValidatorTests
{
    private readonly OrderValidator _orderValidator = new();

    [Fact]
    public void Validate_ValidOrder_DoesNotThrow()
    {
        var order = new Order
        {
            Items =
            [
                new OrderItem { Name = "Widget", Quantity = 1, Price = 10.00m }
            ]
        };

        _orderValidator.Validate(order);
    }

    [Fact]
    public void Validate_OrderWithNoItems_ThrowsException()
    {
        var order = new Order();

        var ex = Assert.Throws<InvalidOperationException>(() => _orderValidator.Validate(order));

        Assert.Equal("Order must contain at least one item.", ex.Message);
    }
}

File: OrderProcessorCpp/tests/UnitTests/OrderValidatorTests.cpp
#include <gtest/gtest.h>
#include "../src/services/OrderValidator.h"
#include "../src/models/order.h"

TEST(OrderValidatorTests, ValidOrder) {
    Order o;
    o.orderId = "ok";
    o.items = { { "sku", 1, 9.99 } };
    EXPECT_TRUE(OrderValidator::Validate(o));
}

TEST(OrderValidatorTests, InvalidOrderEmptyId) {
    Order o;
    o.orderId = "";
    o.items = { { "sku", 1, 9.99 } };
    EXPECT_FALSE(OrderValidator::Validate(o));
}